lib/spack/spack/compiler.py

```diff
@@ -297,11 +297,21 @@ def verify_executables(self):
         Raises a CompilerAccessError if any of the non-null paths for the
         compiler are not accessible.
         """
-        missing = [cmp for cmp in (self.cc, self.cxx, self.f77, self.fc)
-                   if cmp and not (os.path.isfile(cmp) and
-                                   os.access(cmp, os.X_OK))]
-        if missing:
-            raise CompilerAccessError(self, missing)
+        def accessible_exe(exe):
+            # compilers may contain executable names (on Cray or user edited)
+            if not os.path.abspath(exe):
```

Comment:
`os.path.abspath` returns an absolute path when given a path (which may be relative or absolute). When I give it a name like "foo" (i.e. a relative path referring to the local directory) which does not exist, it appears to create an absolute path (which consists of the CWD and that name). In other words, I think this check will fail. Let me look at the logic prior to this edit to suggest another approach.

Reply:
I think `which_string` by itself is sufficient for checking things as in

```
def accessible_exe(exe):
  return bool(which_string(exe))
```

Reply:
That's because I wrote `abspath` when I meant `isabs`. I'll slow down and make sure I didn't do anything else equally silly.